libs/features/seotech/seotech.js

```diff
@@ -21,14 +21,45 @@ export async function getVideoObject(url, seotechAPIUrl) {
   return body.videoObject;
 }
 
-export default async function appendVideoObjectScriptTag(url, { createTag, getConfig }) {
+export async function getStructuredData(url, sheetUrl, seotechAPIUrl) {
+  const apiUrl = new URL(seotechAPIUrl);
+  apiUrl.pathname = '/api/v1/web/seotech/getStructuredData';
+  apiUrl.searchParams.set('url', url);
+  if (sheetUrl) {
+    apiUrl.searchParams.set('sheetUrl', sheetUrl);
+  }
+  const resp = await fetch(apiUrl.href, { headers: { 'Content-Type': 'application/json' } });
+  const body = await resp?.json();
+  if (!resp.ok) {
+    throw new Error(`Failed to fetch structured data: ${body?.error}`);
+  }
+  return body.objects;
+}
+
+export async function appendScriptTag({ locationUrl, getMetadata, createTag, getConfig }) {
+  const windowUrl = new URL(locationUrl);
   const seotechAPIUrl = getConfig()?.env?.name === 'prod'
     ? SEOTECH_API_URL_PROD : SEOTECH_API_URL_STAGE;
-  try {
-    const obj = await getVideoObject(url, seotechAPIUrl);
+
+  const append = (obj) => {
     const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(obj));
     document.head.append(script);
-  } catch (e) {
-    logError(e.message);
+  };
+
+  const prs = [];
+  if (getMetadata('seotech-structured-data') === 'on') {
+    const pageUrl = `${windowUrl.origin}${windowUrl.pathname}`;
+    const sheetUrl = (new URLSearchParams(windowUrl.search)).get('seotech-sheet-url') || getMetadata('seotech-sheet-url');
+    prs.push(getStructuredData(pageUrl, sheetUrl, seotechAPIUrl)
+      .then((r) => r.forEach((obj) => append(obj)))
+      .catch((e) => logError(e.message)));
   }
+  if (getMetadata('seotech-video-url')) {
+    prs.push(getVideoObject(getMetadata('seotech-video-url'), seotechAPIUrl)
+      .then((r) => append(r))
+      .catch((e) => logError(e.message)));
+  }
+  return Promise.all(prs);
```

Comment:
Is there a need to further chain this in the calling function?

Reply:
Yes? Caller is `then` of another promise in impl. We also `await` in tests.